quickwit/quickwit-ingest/src/ingest_v2/ingester.rs

```diff
@@ -156,9 +196,14 @@ impl Ingester {
         for queue_id in queue_ids {
             append_eof_record_if_necessary(&mut state_guard.mrecordlog, &queue_id).await;
 
-            let solo_shard = SoloShard::new(ShardState::Closed, Position::Eof);
-            let shard = IngesterShard::Solo(solo_shard);
-            state_guard.shards.insert(queue_id.clone(), shard);
+            let indexing_status = if is_queue_drained(&state_guard.mrecordlog, &queue_id) {
+                IndexingStatus::Complete
```

Comment:
Isn't this a shift in the semantics of indexing status?
Since it is ambiguous, can we have its semantics as a comment on the enum

Reply:
Ok investigating it is a bit difficult.

So in the ingester:
`IndexingStatus::Complete` means the queue has entirely been consumed. The signal is coming from the truncation.

In the ingester source:
`IndexingStatus::Complete` means the EOF record has been met. This is a reversible state. For instance if the pipeline fails.

Reply:
It does not have to be called `IndexingStatus` here. I was hoping to reuse the enum that you're about to move in `quickwit-proto` but I agree that it's not the same thing from a semantics perspective.